transport/src/main/java/io/netty/channel/DefaultFileRegion.java

```diff
@@ -168,24 +183,55 @@ protected void deallocate() {
     }
 
     @Override
-    public FileRegion retain() {
-        super.retain();
+    public int refCnt() {
+        return refCnt.refCnt();
+    }
+
+    @Override
+    public boolean release() {
+        return refCnt.release();
+    }
+
+    @Override
+    public boolean release(int decrement) {
+        return refCnt.release(decrement);
+    }
+
+    @Override
+    public DefaultFileRegion retain() {
+        refCnt.retain();
         return this;
     }
 
     @Override
-    public FileRegion retain(int increment) {
-        super.retain(increment);
+    public DefaultFileRegion retain(int increment) {
```

Comment:
One more question, is this because we may break the code which extends DefaultFileRegion, or that we want to maintain drop in replacement between patch releases of netty since the JVM may include the return value to generate the method signature? I ask this because the Native.sendfile uses DefaultFileRegion as the parameter and I want to change it to AbstractFileRegion if possible in the follow on PRs which add native support for SlicedFileRegion.